### PR TITLE
MINOR KAFKA-7406: Follow up and address final comments

### PR DESCRIPTION
**Pull request by bbejeck**
Addressing final comments from KAFKA-7406

I used the existing tests for verification.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
@guozhangwang @mjsax

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java [hunk 1]
```diff
@@ -892,8 +892,8 @@ void process(final ProcessorSupplier<? super K, ? super V> processorSupplier,
      * <p>
      * Because a new key is selected, an internal repartitioning topic may need to be created in Kafka if a
      * later operator depends on the newly selected key.
-     * This topic will be as "${applicationId}-XXX-repartition", where "applicationId" is user-specified in
-     * {@link  StreamsConfig} via parameter {@link StreamsConfig#APPLICATION_ID_CONFIG APPLICATION_ID_CONFIG}, "XXX" is
+     * This topic will be as "${applicationId}-&lt;name&gt-repartition", where "applicationId" is user-specified in
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java:895]: We used `${..}` to signify parameter substitution for applicationId. Should we do the same for name?

**Review by vvcephei — APPROVED** [review]
LGTM. One comment.

File: streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
  * The class that is used to capture the key and value {@link Serde}s and set the part of name used for
  * repartition topics when performing {@link KStream#groupBy(KeyValueMapper, Grouped)}, {@link
  * KStream#groupByKey(Grouped)}, or {@link KTable#groupBy(KeyValueMapper, Grouped)} operations.  Note
- * that Kafka Streams does not always create repartition topic for grouping operations.
+ * that Kafka Streams does not always create repartition topic[s] for grouping operations.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java]: I used `[]` in the comment only to highly the change... `[s]` -> `s`

File: streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java [hunk 3]
```diff
@@ -109,10 +109,11 @@ protected Grouped(final Grouped<K, V> grouped) {
 
 
     /**
-     * Create a {@code Grouped} instance with the provided {@code keySerde} and {@code valueSerde}.
+     * Create a {@code Grouped} instance with the provided keySerde and valueSerde.  If the keySerde and/or the valueSerde is
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java]: `@code` -> `@link`

File: streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java [hunk 4]
```diff
@@ -125,10 +126,10 @@ protected Grouped(final Grouped<K, V> grouped) {
 
     /**
      * Perform the grouping operation with the name for a repartition topic if required.  Note
-     * that Kafka Streams does not always create a repartition topic for grouping operations.
+     * that Kafka Streams does not always create repartition topic[s] for grouping operations.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java]: as above

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java:895]: Maybe -- however, application.id is a configuration parameter and cannot be specified on the method while name is a local parameter (my idea was to use different markup to distinguish both)

Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java:895]: I'm inclined to agree with @mjsax as it's a local parameter.  @vvcephei if you insist I can change it.

File: streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java (hunk 3 above)
Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java]: ack

File: streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java (hunk 4 above)
Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/kstream/Grouped.java]: ack

**Comment by bbejeck** [comment]
@vvcephei and @mjsax addressed comments

**Review by mjsax — APPROVED** [review]
LGTM.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java (hunk 1 above)
Comment (vvcephei) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java:895]: Ok, that's fine with me!

**Pull request merged**